Source/Particles/Collision/BinaryCollision/PhotonPhoton/SinglePhotonPhotonCollisionEvent.H

```diff
@@ -0,0 +1,183 @@
+    //     x_star denotes the value of x in the center of mass frame
+
+    using namespace amrex::literals;
+
+    const amrex::ParticleReal w_min = amrex::min(w1, w2);
+    const amrex::ParticleReal w_max = amrex::max(w1, w2);
+
+    constexpr auto one_pr = amrex::ParticleReal(1.);
+    constexpr auto two_pr = amrex::ParticleReal(2.);
+    constexpr auto one_half_pr = amrex::ParticleReal(1./2.);
+    constexpr amrex::ParticleReal c = PhysConst::c;
+    constexpr amrex::ParticleReal c_sq = c*c;
+    constexpr amrex::ParticleReal me = PhysConst::m_e;
+    auto constexpr pow2 = [](double const x) { return x*x; };
+
+    // Compute momenta in the lab frame
+    const amrex::ParticleReal p1x = u1x * me;
+    const amrex::ParticleReal p1y = u1y * me;
+    const amrex::ParticleReal p1z = u1z * me;
+    const amrex::ParticleReal p2x = u2x * me;
+    const amrex::ParticleReal p2y = u2y * me;
+    const amrex::ParticleReal p2z = u2z * me;
+    const amrex::ParticleReal p1 = std::sqrt(pow2(p1x)+pow2(p1y)+pow2(p1z));
+    const amrex::ParticleReal p2 = std::sqrt(pow2(p2x)+pow2(p2y)+pow2(p2z));
+
+    // Compute Lorentz factor gamma in the lab frame
+    const amrex::ParticleReal g1 = p1 / ( me * c );
+    const amrex::ParticleReal g2 = p2 / ( me * c );
+
+    // Compute the cosine of the angle between the two photon momenta in the lab frame
+    const amrex::ParticleReal cos_ang = (p1x*p2x+p1y*p2y+p1z*p2z)/(p1*p2);
+
+    // Energy squared of each and every particle involved in the collision
+    // (both colliding photons and the produced electron and positron)
+    // in the center of momentum frame (they are all equal),
+    // calculated using the Lorentz invariance of the total four-momentum norm
+    const amrex::ParticleReal E_star_sq = one_half_pr*c_sq*p1*p2*(one_pr - cos_ang);
+
+    // Kinetic energy of each particle in the center of momentum frame
+    const amrex::ParticleReal E_star = std::sqrt(E_star_sq);
+
+    // Compute linear Breit Wheeler cross section as a function of the kinetic energy
+    // in the center of momentum frame
+    auto lbw_cross_section = amrex::ParticleReal(0.);
+    lbw_cross_section = LinearBreitWheelerCrossSection(E_star);
+
+    // Lorentz factors in the center of momentum frame
+    const amrex::ParticleReal g1_star = E_star / (me*c_sq);
+    const amrex::ParticleReal g2_star = g1_star;
+
+    // relative velocity in the center of momentum frame
+    const amrex::ParticleReal v_rel = two_pr*c;
+
+    // Cross section and relative velocity are computed in the center of momentum frame.
+    // On the other hand, the particle densities (weight over volume) in the lab frame are used. To
+    // take into account this discrepancy, we need to multiply the collision probability by the ratio
+    // between the Lorentz factors in the COM frame and the Lorentz factors in the lab frame
+    // (see Perez et al., Phys.Plasmas.19.083104 (2012))
+    const amrex::ParticleReal lab_to_COM_factor = g1_star*g2_star/(g1*g2);
```

Comment:
It's not trivial to me that this still works in the case of photons.

According to the paper cited here, the Lorentz invariant quantity is the product sigma\*v_rel\*g1\*g2.

For massive particles, we compute sigma*v_rel in the COM frame and then convert this quantity to the lab frame by multiplying it by g1_star\*g2_star/(g1\*g2).

In the case of photons, the Lorentz factors are defined as those that electrons with the same momentum as the photons would have. So I think that to have a Lorentz invariant quantity, v_rel in the product above should also be the relative velocity that electrons with the same momentum as the photons would have.

In this case, one way to compute sigma\*v_rel in the lab frame would be to:
- compute v_rel directly in the lab frame. (I think it's c\*sqrt(2\*(1-cos(theta))) but needs to be reverified)
- compute sigma in the COM frame as done here but then convert to the lab frame by multiplying it by g1_star\*g2_star\*v_rel_e_star/(g1\*g2\*v_rel_e), where the v_rel_e are the ones that electrons with the same momentum as the photons would have.

Do you think that would work? Maybe we could also discuss if there are better ways to do this.